pkg/addonmanager/controllers/clustermanagement/controller.go

```diff
@@ -126,6 +126,13 @@ func (c *clusterManagementController) sync(ctx context.Context, syncCtx factory.
 		return utils.PatchAddonCondition(ctx, c.addonClient, addonCopy, addon)
 	}
 
+	// Remove unsupported configuration condition if configuration becomes corrected
+	unsupportedConfigCondition := meta.FindStatusCondition(addon.Status.Conditions, UnsupportedConfigurationType)
+	if unsupportedConfigCondition != nil {
+		meta.RemoveStatusCondition(&addonCopy.Status.Conditions, UnsupportedConfigurationType)
```

Comment:
why we remove this condition instead of setting it to false?

Reply:
updated, it better to set it to false than removing it